x-pack/filebeat/input/cometd/input_test.go

```diff
@@ -31,266 +33,295 @@ const (
-		require.NoError(t, err)
-		require.NotNil(t, input)
+	input2, err := NewInput(cfg2, connector, inputContext)
+	require.NoError(t, err)
+	require.NotNil(t, input2)
 
-		// run input
-		input.Run()
+	require.Equal(t, 0, bay.GetConnectedCount())
+	// run input
+	input1.Run()
+	defer input1.Stop()
 
-		go func() {
-			time.Sleep(time.Second) // let input.Stop() be executed.
-			input.Wait()
-		}()
+	// run input
+	input2.Run()
+	defer input2.Stop()
 
-		start := time.Now()
-		for range []beat.Event{<-eventsCh} {
-			if time.Since(start) > time.Second {
-				require.Fail(t, "Channel was not closed after Wait()")
-			}
-		}
-	})
-
-	t.Run("Check input stop", func(t *testing.T) {
-		conf := defaultConfig()
-		logger := logp.NewLogger("test")
-		authParams := bay.AuthenticationParameters{}
-		inputCtx, cancelInputCtx := context.WithCancel(context.Background())
-		workerCtx, workerCancel := context.WithCancel(inputCtx)
-		defer cancelInputCtx()
-
-		input := &cometdInput{
-			config:       conf,
-			log:          logger,
-			inputCtx:     inputCtx,
-			workerCtx:    workerCtx,
-			workerCancel: workerCancel,
-			authParams:   authParams,
-		}
-		input.msgCh = make(chan bay.MaybeMsg)
+	for _, event := range []beat.Event{<-eventsCh, <-eventsCh} {
+		channel, err := event.GetValue("cometd.channel_name")
+		require.NoError(t, err)
 
-		input.Stop()
-		select {
-		case <-workerCtx.Done():
-		default:
-			require.NoError(t, fmt.Errorf("input is not stopped."))
-		}
-	})
-
-	t.Run("Check input wait", func(t *testing.T) {
-		conf := defaultConfig()
-		logger := logp.NewLogger("test")
-		authParams := bay.AuthenticationParameters{}
-		inputCtx, cancelInputCtx := context.WithCancel(context.Background())
-		workerCtx, workerCancel := context.WithCancel(inputCtx)
-		defer cancelInputCtx()
-
-		input := &cometdInput{
-			config:       conf,
-			log:          logger,
-			inputCtx:     inputCtx,
-			workerCtx:    workerCtx,
-			workerCancel: workerCancel,
-			authParams:   authParams,
+		if channel == "channel_name1" {
+			assertEventMatches(t, expected1, event)
+		} else {
+			assertEventMatches(t, expected2, event)
 		}
-		input.msgCh = make(chan bay.MaybeMsg)
+	}
+}
 
-		go func() {
-			input.Wait()
-		}()
+func TestStop(t *testing.T) {
+	conf := defaultConfig()
+	logger := logp.NewLogger("test")
+	authParams := bay.AuthenticationParameters{}
+	inputCtx, cancelInputCtx := context.WithCancel(context.Background())
+	workerCtx, workerCancel := context.WithCancel(inputCtx)
+	defer cancelInputCtx()
+	input := &cometdInput{
+		config:       conf,
+		log:          logger,
+		inputCtx:     inputCtx,
+		workerCtx:    workerCtx,
+		workerCancel: workerCancel,
+		authParams:   authParams,
+	}
+	input.msgCh = make(chan bay.MaybeMsg)
 
-		select {
-		case <-workerCtx.Done():
-		case <-time.After(time.Second): // let input.Stop() be executed.
-			require.NoError(t, fmt.Errorf("input is not stopped."))
-		}
-	})
+	input.Stop()
+	select {
+	case <-workerCtx.Done():
+	default:
```

Comment:
```suggestion
	case <-time.After(time.Second): // let input.Stop() be executed.
```

This could use the same timeout logic that you use in `TestWait` below.

Reply:
Right.